src/lit-html.ts

```diff
@@ -626,7 +626,10 @@ export class TemplateInstance {
   }
 
   _clone(): DocumentFragment {
-    const fragment = document.importNode(this.template.element.content, true);
+    // Clone the node, rather than importing it, to keep the fragment in the
+    // template's document. This leaves the fragment inert so custom elements
+    // won't upgrade until after the main document adopts the node.
```

Comment:
When does the main document adopt the clone? This seems like it will always leave the elements inert.

Reply:
Nodes are adopted by a document when they get added somewhere in that document's node tree. In the case of lit-html that is the `render` function at

https://github.com/Polymer/lit-html/blob/f4e8db4509d41d78b2d18563c0b0831d62f0816f/src/lit-html.ts#L184